packages/flutter_migrate/lib/src/commands/resolve_conflicts.dart

```diff
@@ -0,0 +1,400 @@
+      // We write a newline in the output, this counteracts it.
+      if (lines.last == '') {
+        lines.removeLast();
+      }
+
+      // Find all conflicts
+      final List<Conflict> conflicts = findConflicts(lines, localPath);
+
+      // Prompt developer
+      final CommandResult? promptResult =
+          await promptDeveloperSelectAction(conflicts, lines, localPath);
+      if (promptResult != null) {
+        return promptResult;
+      }
+
+      final bool result =
+          await verifyAndCommit(conflicts, lines, file, localPath);
+      if (!result) {
+        i--;
+      }
+    }
+    return const CommandResult(ExitStatus.success);
+  }
+
+  /// Parses the lines of a file and extracts a list of Conflicts.
+  List<Conflict> findConflicts(List<String> lines, String localPath) {
+    // Find all conflicts
+    final List<Conflict> conflicts = <Conflict>[];
+    Conflict currentConflict = Conflict.empty();
+    for (int lineNumber = 0; lineNumber < lines.length; lineNumber++) {
+      final String line = lines[lineNumber];
+      if (line.contains(_conflictStartMarker)) {
+        currentConflict.startLine = lineNumber;
+      } else if (line.contains(_conflictDividerMarker)) {
+        currentConflict.dividerLine = lineNumber;
+      } else if (line.contains(_conflictEndMarker)) {
+        currentConflict.endLine = lineNumber;
+        if (!(currentConflict.startLine == null ||
+                currentConflict.dividerLine == null ||
+                currentConflict.startLine! < currentConflict.dividerLine!) ||
+            !(currentConflict.dividerLine == null ||
+                currentConflict.endLine == null ||
+                currentConflict.dividerLine! < currentConflict.endLine!)) {
+          throw StateError(
+              'Invalid merge conflict detected in $localPath: Improperly ordered conflict markers.');
+        }
+        conflicts.add(currentConflict);
+        currentConflict = Conflict.empty();
+      }
+    }
+    return conflicts;
+  }
+
+  /// Display a detected conflict and prompt the developer on whether to accept the original lines, new lines,
+  /// or skip handling the conflict.
+  Future<CommandResult?> promptDeveloperSelectAction(
+      List<Conflict> conflicts, List<String> lines, String localPath) async {
+    final int contextLineCount = int.parse(stringArg('context-lines')!);
+    for (final Conflict conflict in conflicts) {
+      if (!conflict.isValid) {
+        conflict.selection = ConflictSelection.skip;
+        continue;
+      }
+      // Print the conflict for reference
+      logger.printStatus(terminal.clearScreen(), newline: false);
+      logger.printStatus('Cyan', color: TerminalColor.cyan, newline: false);
+      logger.printStatus(' = Original lines.  ', newline: false);
+      logger.printStatus('Green', color: TerminalColor.green, newline: false);
+      logger.printStatus(' = New lines.\n', newline: true);
+
+      // Print the conflict for reference
```

Comment:
Nit: Please scrub the PR for missing periods on comments; it's currently inconsistent.